### DISCUSSION TITLE
혹시 포스트프로세싱을 코드로 제어할수는 없나요?

### DISCUSSION THREAD

**Discussion by ted002** [naverz/zepeto-studio-kor#469] [5. Q&A]
현재 포스트프로세싱을 쭉 조정해보고 있는데 ,코드로 포스트 프로세싱을 조정하려고 하니 문제가 생기네요.

유니티의 예제로 포스트 프로세싱을 코드로 조정해보려고 하는데, 계속 오류가 뜹니다. 

거의 대다수의 경우에서 내부인자를 조정할수 없다고 하는데, 어떻게 하면 조정할수 있나요? 혹시 코드로는 조정할수 없게 되어있나요?

**Comment by minjujuu** [comment]
코드를 통해 아래와 같은 기능을 제어하는 것은 가능한 것 같아요~
- 포스트 프로세싱 profile의 Setting 활성화/비활성화
- 포스트 프로세싱 profile에 특정 Setting 추가하기
- 포스트 프로세싱 profile에서 특정 Setting 지우기

그 외에 intensity, color 등을 코드에서 조절하는 거는 저도 해보고 있는데, 방법을 찾게 되면 공유드리겠습니다!
저는 우선 profile을 새로 하나 더 만들어서 교체하는 방식으로 진행하고 있어요.
```
    public postProcessingVolume: PostProcessVolume;
    public switchProfile: PostProcessProfile;
    Start() {
            let profile = this.postProcessingVolume.profile;

            // profile 교체
            profile = this.switchProfile;
 
            // Setting 추가 
            profile.AddSettings<Vignette>(); 

            // Setting 삭제
            if(profile.HasSettings<ColorGrading>())
                profile.RemoveSettings<ColorGrading>(); 

            //현재 profile의 모든 Setting을 on/off
            profile.settings.ForEach(
                (setting)=>{
                    setting.active = !setting.active;
                }
            );
      }
```

**Reply by ted002** [reply]
오 감사합니다! 한번 저도 테스트해볼게요!
